### PR TITLE
MWPW-128665  Add Collection Title input field 

### PR DESCRIPTION
**Pull request by cmiqueo**
This PR adds new functionality to the CAAS Configurator so that authors can add a collection title directly in the configurator UI.

The new "Collection Title" input field provides three options: 
- English Title  -  does not need translation nor a placeholder mappings file  
- Placeholder key - for titles that are localized and use placeholder mappings file  
- Empty - for collections without title

See video demo:
https://jira.corp.adobe.com/secure/attachment/9849299/Collection%20Title%20Authoring.mp4

It also adds a "title" attribute to the input field that it is used as a "tooltip"
image

Resolves: [MWPW-128665](https://jira.corp.adobe.com/browse/MWPW-128665)

NOTE: This will not affect existing collection.  This new functionality will only be available in the CAAS Configurator UI.

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/tools/caas?martech=off
- After: https://mwpw-128665--milo--cmiqueo.hlx.page/tools/caas?martech=off

cc: @sanrai, @chrischrischris , @auniverseaway

### CODE REVIEW

**Review by chrischrischris — COMMENTED** [review]
`collectionTitle` also needs to be added to `defaultState` at bottom of caas/utils.js

File: libs/blocks/caas/utils.js [hunk 1]
```diff
@@ -285,7 +285,7 @@ export const getConfig = async (state, strs = {}) => {
         prettyDateIntervalFormat:
           strs.prettyDateIntervalFormat || '{ddd}, {LLL} {dd} | {timeRange} {timeZone}',
         totalResultsText: strs.totalResults || '{total} results',
-        title: strs.collectionTitle || '',
+        title: state.collectionTitle?.match(/^{.*}$/) ? strs[state.collectionTitle.replace(/{|}/g, '')] : state.collectionTitle || 'My Awesome Title',
```
[the remark this answers is not in the document]
Comment (cmiqueo) [reply libs/blocks/caas/utils.js]: @chrischrischris Thanks for checking. It's all good now.

File: libs/blocks/caas-config/caas-config.js [hunk 2]
```diff
@@ -309,6 +310,8 @@ const BasicsPanel = ({ tagsData }) => {
 
   return html`
     <${Input} label="Collection Name (only displayed in author link)" prop="collectionName" type="text" />
+    <${Input} label="Collection Title" prop="collectionTitle" type="text" title="Enter a title, {placeholder}, or leave empty "/>
+
```
Comment (meganthecoder) [inline libs/blocks/caas-config/caas-config.js:314]: Remove extra line

File: test/blocks/caas/utils.test.js [hunk 3]
```diff
@@ -164,7 +164,7 @@ describe('getConfig', () => {
           cardTitleAccessibilityLevel: 6,
           prettyDateIntervalFormat: '{ddd}, {LLL} {dd} | {timeRange} {timeZone}',
           totalResultsText: '{total} Results',
-          title: 'My Awesome Title',
+          title: '',
```
Comment (meganthecoder) [inline test/blocks/caas/utils.test.js:167]: It looks like this only tests one of your options for the title (AKA "leave empty"). Do you want to add tests for the other two?

**Review by meganthecoder — COMMENTED** [review]
Couple comments. Also, I'm curious about why you aren't using a milo link for your "After" test url

**Comment by cmiqueo** [comment]
> Couple comments. Also, I'm curious about why you aren't using a milo link for your "After" test url

I don't have access :(

**Comment by meganthecoder** [comment]
> > Couple comments. Also, I'm curious about why you aren't using a milo link for your "After" test url
> 
> I don't have access :(

 @chrischrischris How do we handle testing links and verification for people without access?

**Comment by sukamat** [comment]
> > > Couple comments. Also, I'm curious about why you aren't using a milo link for your "After" test url
> > 
> > 
> > I don't have access :(
> 
> @chrischrischris How do we handle testing links and verification for people without access?

Isn't that just about changing the URL to point to the branch and forked repo name? Or is there more involved in this ticket that needs additional access?

https://mwpw-128665--milo--cmiqueo.hlx.page/tools/caas?martech=off

File: test/blocks/caas/utils.test.js (hunk 3 above)
Comment (cmiqueo) [reply test/blocks/caas/utils.test.js:167]: @meganthecoder thanks for checking. :)
Other collection title options are tested here:
https://github.com/adobecom/milo/blob/e4455917032bfa9a2e89787ae23de8d885d6e2ae/test/blocks/caas/utils.test.js#L73-L96

**Review by chrischrischris — APPROVED** [review]

**Review by sukamat — APPROVED** [review]

**Review by meganthecoder — APPROVED** [review]

**Review by honstar — APPROVED** [review]

**Pull request merged**